pkg/media/opuswriter/opuswriter.go

```diff
@@ -0,0 +1,228 @@
+		channelCount:  channelCount,
+		serial:        serial,
+		checksumTable: make([]int, 256),
+	}
+	writer.initChecksum()
+	if err := writer.writeHeaders(); err != nil {
+		return nil, err
+	}
+
+	return writer, nil
+}
+
+/*
+    ref: https://tools.ietf.org/html/rfc7845.html
+    https://git.xiph.org/?p=opus-tools.git;a=blob;f=src/opus_header.c#l219
+
+       Page 0         Pages 1 ... n        Pages (n+1) ...
+    +------------+ +---+ +---+ ... +---+ +-----------+ +---------+ +--
+    |            | |   | |   |     |   | |           | |         | |
+    |+----------+| |+-----------------+| |+-------------------+ +-----
+    |||ID Header|| ||  Comment Header || ||Audio Data Packet 1| | ...
+    |+----------+| |+-----------------+| |+-------------------+ +-----
+    |            | |   | |   |     |   | |           | |         | |
+    +------------+ +---+ +---+ ... +---+ +-----------+ +---------+ +--
+    ^      ^                           ^
+    |      |                           |
+    |      |                           Mandatory Page Break
+    |      |
+    |      ID header is contained on a single page
+    |
+    'Beginning Of Stream'
+
+   Figure 1: Example Packet Organization for a Logical Ogg Opus Stream
+*/
+
+func (i *OpusWriter) writeHeaders() error {
+	// ID Header
+	oggIDHeader := make([]byte, 19)
+
+	copy(oggIDHeader[0:], []byte("OpusHead"))                     // Magic Signature 'OpusHead'
+	oggIDHeader[8] = 1                                            // Version
+	oggIDHeader[9] = uint8(i.channelCount)                        // Channel count
+	binary.LittleEndian.PutUint16(oggIDHeader[10:], 0)            // pre-skip, don't need to skip any value
+	binary.LittleEndian.PutUint32(oggIDHeader[12:], i.sampleRate) // original sample rate, any valid sample e.g 48000
+	binary.LittleEndian.PutUint16(oggIDHeader[16:], 0)            // output gain
+	oggIDHeader[18] = 0                                           // channel map 0 = one stream: mono or stereo
+
+	// Reference: https://tools.ietf.org/html/rfc7845.html#page-6
+	// RFC specifies that the ID Header page should have a granule position of 0 and a Header Type set to 2 (StartOfStream)
+	data := i.createPage(oggIDHeader, 2, 0)
+	if _, err := i.stream.Write(data); err != nil {
+		return err
+	}
+
+	// Comment Header
+	oggCommentHeader := make([]byte, 21)
+	copy(oggCommentHeader[0:], []byte("OpusTags"))          // Magic Signature 'OpusTags'
+	binary.LittleEndian.PutUint32(oggCommentHeader[8:], 5)  // Vendor Length
+	copy(oggCommentHeader[12:], []byte("pions"))            // Vendor name 'pions'
+	binary.LittleEndian.PutUint32(oggCommentHeader[17:], 0) // User Comment List Length
+
+	// RFC specifies that the page where the CommentHeader completes should have a granule position of 0
+	data = i.createPage(oggCommentHeader, 0, 0)
+	if _, err := i.stream.Write(data); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// CRC32
+func (i *OpusWriter) initChecksum() {
```

Comment:
https://golang.org/pkg/hash/crc32

This looks like `crc32.MakeTable(0x04c11db7)` although the implementation is slightly different but probably the same value.